client/client.go

```diff
@@ -176,6 +176,11 @@ func NewBaseClient(credentials *Credentials, absolutePath string, isHTTP bool) (
 
 // NewRequest creates a request
 func (c *Client) NewRequest(ctx context.Context, method, urlStr string, body interface{}) (*http.Request, error) {
+	// check if client exists or not
+	if c.client == nil {
+		return nil, fmt.Errorf("client for %s is missing", c.UserAgent)
```

Comment:
Improve the error message. User should get an idea on what parameters he is missing.